scylla-cql/src/types/deserialize/row.rs

```diff
@@ -0,0 +1,101 @@
+//! Provides types for dealing with row deserialization.
+
+use super::FrameSlice;
+use crate::frame::frame_errors::ParseError;
+use crate::frame::response::result::ColumnSpec;
+
+/// Represents a raw, unparsed column value.
+#[non_exhaustive]
+pub struct RawColumn<'frame> {
+    pub spec: &'frame ColumnSpec,
```

Comment:
This is how @piodul designed the framework.

Reply:
With prepared statement metadata caching IMO it's not worth making the column specs lazily deserializable.

Reply:
Metadata caching is disable by default because of the issue with `SELECT *`, isn't it?

Reply:
> Metadata caching is disable by default because of the issue with `SELECT *`, isn't it?

It is. I decided to disable it by default for two reasons:
1. rust driver issue - we decided to leave metadata immutable. Performance-aware users should not alter the schema after enabling this optimisation.
2. cql protocol issue - the multi client scenario (related to `SELECT *`), where only one client reprepares the statement. The other clients are not notified about the repreparation and so they store outdated metadata. To reproduce this issue, the rust driver issue mentioned above needs to be solved firstly.

Reply:
> It is. I decided to disable it by default for two reasons:

    rust driver issue - we decided to leave metadata immutable. Performance-aware users should not alter the schema after enabling this optimisation.
    cql protocol issue - the multi client scenario (related to SELECT *), where only one client reprepares the statement. The other clients are not notified about the repreparation and so they store outdated metadata. To reproduce this issue, the rust driver issue mentioned above needs to be solved firstly (because, as of now, the driver doesn't update the metadata anyway).

This should definitely be put in the docstring of the optimisation setting, so that it is clear both for us and for users.

Reply:
> > It is. I decided to disable it by default for two reasons:
> 
> > 1. rust driver issue - we decided to leave metadata immutable. Performance-aware users should not alter the schema after enabling this optimisation.
> > 2. cql protocol issue - the multi client scenario (related to SELECT *), where only one client reprepares the statement. The other clients are not notified about the repreparation and so they store outdated metadata. To reproduce this issue, the rust driver issue mentioned above needs to be solved firstly (because, as of now, the driver doesn't update the metadata anyway).
> 
> This should definitely be put in the docstring of the optimisation setting, so that it is clear both for us and for users.

I agree. I can take care of it.